Plandomizer.py

```diff
@@ -737,13 +783,22 @@ def get_item(self, ignore_pools, item_pools, location, player_id, record, worlds
         return item
 
     def cloak(self, worlds, location_pools, model_pools):
-        for (name, record) in pattern_dict_items(self.locations):
+        # TODO: Handle lists of models?
+        for (name, record, was_pattern) in pattern_dict_items(self.locations):
             if record.model is None:
                 continue
 
             player_id = self.id if record.player is None else record.player - 1
             world = worlds[player_id]
 
+            skip = True
+            for pool in location_pools:
+                for location in pool:
+                    if location.name == name:
+                        skip = False
+            if skip:
+                continue
+
```

Comment:
This could be condensed into
```suggestion
            if not any(location.name == name for location in itertools.chain.from_iterable(location_pools)):
                continue

```